libs/schedulers/include/hpx/schedulers/local_priority_queue_scheduler.hpp

```diff
@@ -882,6 +882,28 @@ namespace hpx { namespace threads { namespace policies {
             return count;
         }
 
+        // Queries the current number of idle threads (cores)
+        std::int64_t get_idle_thread_count() const override
+        {
+            // Return thread count of one specific queue.
+            std::int64_t count = 0;
+
+            // Return the cumulative count for all queues.
+            for (std::size_t i = 0; i != num_high_priority_queues_; ++i)
+            {
+                count += high_priority_queues_[i].data_->get_thread_count(
+                             unknown) == 0;
+            }
+            count += low_priority_queue_.get_thread_count(unknown) == 0;
+
+            for (std::size_t i = 0; i != num_queues_; ++i)
+            {
+                count += queues_[i].data_->get_thread_count(unknown) == 0;
+            }
+
+            return count;
+        }
```

Comment:
Is this a good idea? It'll return `num_threads * 2 + 1` when all the cores are idle, and at least `2` if one of the cores are idle. Would it be reasonable to to combine the normal and high priority queues (add one if both are empty), and ignore the low priority queue completely?

Reply:
Fair points!

Reply:
Now it returns 1 only if all the normal and high priority queus are empty, and 0 otherwise... ;)

Something like this?
```
            std::int64_t count = 0;

            for (std::size_t i = 0; i != (std::max)(num_queues_, num_high_priority_queues_); ++i)
            {
                bool idle = true;
                if (i < num_queues_)
                    idle &&= queues_[i].data_->get_thread_count(unknown) == 0;
                if (i < num_high_priority_queues_)
                    idle &&= high_priority_queues_[i].data_->get_thread_count(unknown) == 0;

                if (idle) ++count;
            }

            return count;
```